x-pack/plugins/lens/public/shared_components/axis/extent/helpers.ts

```diff
@@ -86,9 +103,39 @@ export function getDataBounds(
   }
 }
 
-export function validateExtent(shouldIncludeZero: boolean, extent?: UnifiedAxisExtentConfig) {
-  return {
-    inclusiveZeroError: shouldIncludeZero && !validateZeroInclusivityExtent(extent),
-    boundaryError: !validateAxisDomain(extent),
-  };
+export function validateExtent(
+  hasBarOrArea: boolean,
+  extent: UnifiedAxisExtentConfig,
+  scaleType?: YScaleType | XScaleType
+): {
+  helpMsg?: string;
+  errorMsg?: string;
+} {
+  if (!validateAxisDomain(extent)) {
+    return {
+      errorMsg: i18n.translate('xpack.lens.axisExtent.boundaryError', {
+        defaultMessage: 'Lower bound has to be less than upper bound',
+      }),
+    };
+  }
+
+  if (scaleType === 'log') {
+    const key = validateLogarithmicExtent(extent) ? 'helpMsg' : 'errorMsg';
+    return {
+      [key]: i18n.translate('xpack.lens.axisExtent.logExcludeZero', {
+        defaultMessage: 'Logarithmic bounds must exclude zero',
+      }),
+    };
+  }
+
+  if (hasBarOrArea) {
+    const key = validateZeroInclusivityExtent(extent) ? 'helpMsg' : 'errorMsg';
+    return {
+      [key]: i18n.translate('xpack.lens.axisExtent.inclusiveZero', {
+        defaultMessage: 'Bounds must include zero',
```

Comment:
This is such a confusing message in my opinion (not coming from your PR but maybe we could rethink it when we're here?). Couldn't we extend the logic and change it to `Lower bound should be 0` when the upper bound is >0 and `Upper bound should be 0` when the lower bound is < 0? I think it is less confusing. CC @amyjtechwriter

Reply:
Yeah that could be simpler @dej611 didn't like this wording either. To me it makes sense that the value `0` must be in the provided range but I could see how that would be confusing and isolating upper/lower could be more focus intent. CC @markov00

Reply:
Same as above, "Bounds must contain zero."

Reply:
Done 97c62b179f91e8f93d37cd0cd2b962a6bda89231